subsurface.qrc

```diff
@@ -6,6 +6,7 @@
         <file alias="subsurface-icon">icons/subsurface-icon.png</file>
         <file alias="subsurface-mobile-icon">icons/subsurface-mobile-icon.png</file>
         <file alias="list-add-icon">icons/plus.png</file>
+        <file alias="mirror-icon">icons/mirror-icon.png</file>
```

Comment:
Is the `mirror-icon.svg` file actually needed, seeing that we are only referencing the `mirror-icon.png` file here?

Reply:
Yes, the svg as a vector format is the "source" for the image which is needed if you later want to change the icon or derive others from it.

Reply:
Cool, makes sense.
So if `mirror-icon.png` is just an artefact produced from `mirror-icon.svg`, wouldn't it make more sense to not store the artefact (like we don't store the `*.o` files produced as artefacts during compilation) and instead get the build process to produce the artefacts during the build?

Reply:
Dunno. I would do the conversation with inkscape which is a GUI tool. Ymmv but I would hate it if during build a GUI app would start up and I would have to click something like "export to png". We do similar things for the user manual where we keep the html around that as also an artifact.

Reply:
You do realize that there is a `renderSVGIcon` function in `qthelper.cpp`?

Reply:
Inkscape actually does work on the command line as well: https://wiki.inkscape.org/wiki/Using_the_Command_Line
But I would probably rather use something like `convert` (based on [ImageMagick](https://imagemagick.org/index.php), which is intended for this kind of use and has a powerful set of command line options to control the properties of the output. And it's very common, so probably already installed on most developers' workstations.

Reply:
I had a look into this, and it looks like SVG files can be used directly as the source for `QIcon` - in fact it this is already done in some places:

https://github.com/subsurface/subsurface/blob/d44787a0b33d96541c34b218f40ed69f58c02794/qt-models/divetripmodel.cpp#L198

I have opened a pull request to refactor the existing cases: #3844.

Reply:
Hmmm... I think the quoted code should be changed to
```
static QPixmap &getGlobeIcon()
{
	static std::unique_ptr<QPixmap> icon = []() {
		const IconMetrics &im = defaultIconMetrics();
		return std::make_unique<QPixmap>(QIcon(":globe-icon").pixmap(im.sz_small, im.sz_small));
	}();
	return *icon;
}
```
That said, we had crashes when freeing Qt resources after main() exited. Because Qt memory management is the worst.